### PR TITLE
[WFLY-14968] Collapse if-statements (Security)

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFLY-14968

### CODE REVIEW

File: security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java [hunk 1]
```diff
@@ -323,37 +323,37 @@ public void push(final String securityDomain) {
         RunAs currentRunAs = current.getIncomingRunAs();
         boolean trusted = currentRunAs != null && currentRunAs instanceof RunAsIdentity;
 
-        if (trusted == false) {
+        if (trusted && SecurityActions.remotingContextIsSet()) {
```
Comment (darranl) [inline security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java:326]: I am just quickly scanning the diff so may have missed some lines but this seems to have lost the "trusted==false"?

Comment (boris-unckel) [reply security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java]: Thanks for the quick reply. I saw the failing test and it is resolved now.

**Review by darranl — APPROVED** [review]
FYI this code is planned to be deleted during WildFly 25 development anyway but should be Ok to merge.

TBH I am not a big fan of these kind of changes, I don't really agree about the readability motivation, separate lines make it a lot easier to read one step at a time and also a lot easier to navigate with a debugger.  I know when I am debugging code I am not overly familiar with the last thing I want to hit is a complex if statement with multiple operators including the very thin not operator.  Newer JVMs are much better at reporting the exact location is things like NPEs but older ones when trying to support users remotely if too much happens on a single line in the past it has been very difficult to deduce what may be null.  But this one does seem sufficiently constrained.

**Comment by boris-unckel** [comment]
@darranl Thanks for your review and comments.
> FYI this code is planned to be deleted during WildFly 25 development anyway but should be Ok to merge.

The merge conflict is visible below and I have a local solution prepared, ready to push. I don't want to block other important tests, so I just push 1-2 at a time.

> 
> TBH I am not a big fan of these kind of changes, I don't really agree about the readability motivation, separate lines make it a lot easier to read one step at a time and also a lot easier to navigate with a debugger. ...

Yersay gave me a similar feedback and I'm already reworking all PRs: Separate lines for separate conditions, except OR expressions in parenthesis. Unnecessary parenthesis removed.

**Comment by boris-unckel** [comment]
@bstansberry Hello Brian, the tests for this PR have passed, it is approved. Can you check and merge, please?

**Pull request merged**